metecho/api/sf_run_flow.py

```diff
@@ -47,6 +47,36 @@
 </Package>"""
 
 
+class ScratchOrgError(Exception):
+    pass
+
+
+def handle_sf_error(err, scratch_org=None, originating_user_id=None):
```

Comment:
This looks like something that might be good to implement as a context manager so you can use it like this:

    with scratch_org_cleanup_on_error(scratch_org, originating_user_id):
        # do things that might error